agent/tests/test_lib_aws_sdk_php.c

```diff
@@ -6,12 +6,284 @@
+
+  /*
+   * nr_lib_aws_sdk_php_get_command_arg_value extracts an arg value from the 2nd
+   * argument in the argument list, so we need to have at least 2 args to
+   * extract properly.
+   */
+  tlib_php_engine_create("");
+  tlib_php_request_start();
+
+  tlib_php_request_eval("function one_param($a) { return; }");
+  nr_php_wrap_user_function(NR_PSTR("one_param"), expect_arg_value_null);
+  tlib_php_request_eval("function two_param_valid($a, $b) { return; }");
+  nr_php_wrap_user_function(NR_PSTR("two_param_valid"),
+                            expect_arg_value_not_null);
+  tlib_php_request_eval("function two_param($a, $b) { return; }");
+  nr_php_wrap_user_function(NR_PSTR("two_param"), expect_arg_value_null);
+  tlib_php_request_eval("function no_param() { return;}");
+  nr_php_wrap_user_function(NR_PSTR("no_param"), expect_arg_value_null);
+
+  /*
+   * The function isn't decoding this arg, so it doesn't matter what it is as
+   * long as it exists.
+   */
+  first_arg = tlib_php_request_eval_expr("1");
+
+  /* Valid case.  The wrapper should verify strings match. */
+
+  char* valid_array_args
+      = "array("
+        "    0 => array("
+        "        'QueueUrl' => 'curly_q'"
+        "    )"
+        ")";
+  array_arg = tlib_php_request_eval_expr(valid_array_args);
+  expr = nr_php_call(NULL, "two_param_valid", first_arg, array_arg);
+  tlib_pass_if_not_null("Expression should evaluate.", expr);
+  nr_php_zval_free(&expr);
+  nr_php_zval_free(&array_arg);
+
+  /* Test Invalid Cases*/
+
+  /* Invalid case: only one parameter.  The wrapper should see the null return
+   * value. */
+  expr = nr_php_call(NULL, "one_param", first_arg);
+  tlib_pass_if_not_null("Expression should evaluate.", expr);
+  nr_php_zval_free(&expr);
+
+  /* Invalid case: no parameter.  The wrapper should see the null return value.
+   */
+  expr = nr_php_call(NULL, "no_param");
+  tlib_pass_if_not_null("Expression should evaluate.", expr);
+  nr_php_zval_free(&expr);
+
+  /*
+   *Invalid case: QueueUrl not found in the argument array.  The wrapper should
+   *see the null return value.
+   */
+  char* no_queueurl_arg
+      = "array("
+        "    0 => array("
+        "        'Nope' => 'curly_q'"
+        "    )"
+        ")";
+  array_arg = tlib_php_request_eval_expr(no_queueurl_arg);
+  expr = nr_php_call(NULL, "two_param", first_arg, array_arg);
+  tlib_pass_if_not_null("Expression should evaluate.", expr);
+  nr_php_zval_free(&expr);
+  nr_php_zval_free(&array_arg);
+
+  /*
+   *Invalid case: inner arg in the argument array is not an array.  The wrapper
+   *should see the null return value.
+   */
+  char* arg_in_array_not_array
+      = "array("
+        "    0 => '1'"
+        ")";
+  array_arg = tlib_php_request_eval_expr(arg_in_array_not_array);
+  expr = nr_php_call(NULL, "two_param", first_arg, array_arg);
+  tlib_pass_if_not_null("Expression should evaluate.", expr);
+  nr_php_zval_free(&expr);
+  nr_php_zval_free(&array_arg);
+
```

Comment:
What if the `QueueUrl` key has a value which is not the correct type, like an `array`?

Reply:
Added here: [b63f74f](https://github.com/newrelic/newrelic-php-agent/pull/1003/commits/b63f74fb558b260d03c91d29c90f623fe0918fb3)